app/test_main.py

```diff
@@ -1 +1,52 @@
-# write your code here
+import pytest
+from unittest import mock
+from unittest.mock import MagicMock
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def mock_valid_google_url() -> MagicMock:
+    with mock.patch("app.main.valid_google_url") as mock_func:
+        yield mock_func
+
+
+@pytest.fixture
+def mock_has_internet_connection() -> MagicMock:
+    with mock.patch("app.main.has_internet_connection") as mock_func:
+        yield mock_func
+
+
+def test_can_access_google_page_accessible(
```

Comment:
The test name `test_can_access_google_page_accessible` could be more descriptive. Consider renaming it to something like `test_can_access_google_page_when_accessible` to better convey the test scenario.